modules/core/src/main/scala/sangria/schema/AstSchemaResolver.scala

```diff
@@ -223,7 +225,7 @@ case class GenericDynamicDirectiveResolver[T, A](
     directiveName: String,
     locations: Set[DirectiveLocation.Value] = Set.empty,
     resolve: GenericDynamicDirectiveContext[A] => Option[T])(implicit
-    val marshaller: ResultMarshallerForType[T])
+    val marshaller: ResultMarshallerForType[A])
```

Comment:
Was this a bug?

Reply:
Yes, without this change I would get an error later on:
```
[error] -- [E007] Type Mismatch Error: /Users/jchyb/Documents/workspace/sangria/modules/core/src/main/scala/sangria/schema/ResolverBasedAstSchemaBuilder.scala:681:52 
[error] 681 |                          createDynamicArgs(astDir)(marshaller)))
[error]     |                                                    ^^^^^^^^^^
[error]     |Found:    (marshaller : sangria.marshalling.ResultMarshallerForType[T])
[error]     |Required: sangria.marshalling.ResultMarshallerForType[A$2]
[error]     |
[error]     |where:    A$2 is a type in an anonymous function in method resolveDirectives with bounds 
[error]     |
[error]     | longer explanation available when compiling with `-explain`
```